packages/marko/src/runtime/components/init-components-browser.js

```diff
@@ -252,41 +251,38 @@ function initClientRendered(componentDefs, doc) {
  * This method initializes all components that were rendered on the server by iterating over all
  * of the component IDs.
  */
-function initServerRendered(renderedComponents, doc) {
+function initServerRendered(renderedComponents, doc, runtimeId) {
   var type = typeof renderedComponents;
-  var globalKey = "$";
-  var runtimeId;
+  var globalKey;
 
   if (type !== "object") {
     if (type === "string") {
       runtimeId = renderedComponents;
-      globalKey += runtimeId + "_C";
+      globalKey = runtimeId;
     } else {
-      globalKey += (runtimeId = DEFAULT_RUNTIME_ID) + "C";
+      globalKey = runtimeId = DEFAULT_RUNTIME_ID;
     }
```

Comment:
What do you think about shortening this to:

```js
if (type === "string") {
  runtimeId = renderedComponents;
} else {
  runtimeId = DEFAULT_RUNTIME_ID;
}

globalKey = runtimeId;
```

Or maybe even:

```js
globalKey = runtimeId = (
  type === "string" ? renderedComponents : DEFAULT_RUNTIME_ID
)
```

Reply:
Yeah good catch. I realized that one as well. With the changes I've made `globalKey` and `runtimeId` are one and the same. I think I'll just remove `globalKey`.

Reply:
Done! Realized I could shorten it further, since really the `type` here would either be `string` or `undefined`. I just made it `runtimeId = renderedComponents || DEFAULT_RUNTIME_ID`.

The API for `init` should be one of `init()`, `init("runtimeID")` or `init(serializedComponentDataObject)`.